src/d3d9/d3d9_common_texture.h

```diff
@@ -353,7 +353,7 @@ namespace dxvk {
     D3D9SubresourceBitset& GetUploadBitmask() { return m_needsUpload; }
 
     void SetNeedsUpload(UINT Subresource, bool upload) { m_needsUpload.set(Subresource, upload); }
-    bool NeedsUpload(UINT Subresource) const { return m_needsUpload.get(Subresource); }
+    bool NeedsUpload(UINT Subresource) const { return m_needsUpload.get(Subresource) && m_buffers[Subresource] != nullptr; }
```

Comment:
We still want to mark this as not needing upload anymore + clear dirty region regardless of whether we have a staging buffer to do it from or not.

We should factor that outside of the NeedsUpload call into FlushBuffer